test-data/unit/check-based-infer-function-types.test

```diff
@@ -203,6 +239,32 @@ def f(**kwargs):  # E: Function is missing a type annotation for one or more arg
 [builtins fixtures/dict.pyi]
 
 
+[case testInferOverloadWithPositionalOnly]
+from typing import overload
+
+@overload
+def foo(a: int, /): ...
+@overload
+def foo(a: str, /): ...
+
+def foo(a):
+    reveal_type(a)
+
+@overload
+def bar(x: int, /, b: str): ...
+@overload
+def bar(y: None, /, b: ellipsis): ...
+
+def bar(z, b):
+    reveal_type(z)
+    reveal_type(b)
+[out]
+main:4: error: invalid syntax  [syntax]
```

Comment:
invalid syntax?

Reply:
`/` is invalid syntax < 3.8

Reply:
@DetachHead I think this branch is complete and can be merged, just this thread remaining